setup.py

```diff
@@ -3,37 +3,26 @@
 
 import sys
 import platform
+import toml
 
+from os.path import join, dirname
 from setuptools import setup, find_packages, Extension
 
 
-install_requires = [
-    "ply>=3.4,<4.0",
-    "six~=1.15",
-]
-
-tornado_requires = [
-    "tornado>=4.0,<7.0; python_version>='3.12'",
-    "tornado>=4.0,<6.0; python_version<'3.12'",
-]
+meta = toml.load(join(dirname(__file__), 'pyproject.toml') )
+install_requires = meta["project"]["dependencies"]
+dev_requires = meta["project"]["optional-dependencies"]["dev"]
+tornado_requires = meta["project"]["optional-dependencies"]["tornado"]
 
 try:
     from tornado import version as tornado_version
     if tornado_version < '5.0':
         tornado_requires.append("toro>=0.6")
+        dev_requires.append("toro>=0.6")
```

Comment:
@aisk  Do you think this implementation makes sense now？

Reply:
Tornado support will be deprecated, so I think this is ok for me.